src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -483,9 +483,9 @@ bool looksLikeCast()
             return null;
         }
 
-        private PatternSyntax ParsePattern(Precedence precedence)
+        private PatternSyntax ParsePattern(Precedence precedence, bool whenIsKeyword = false)
         {
-            var node = ParseExpressionOrPattern(forCase: false, precedence: precedence);
+            var node = ParseExpressionOrPattern(whenIsKeyword: whenIsKeyword, precedence: precedence);
```

Comment:
Probably don't need the named arguments, since names are clear and types are distincts.